pkg/collector/collector.go

```diff
@@ -0,0 +1,237 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"net"
+	"net/url"
+	"os"
+	"time"
+
+	eraserv1alpha1 "github.com/Azure/eraser/api/v1alpha1"
+	"github.com/Azure/eraser/pkg/logger"
+	"google.golang.org/grpc"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	pb "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	// unixProtocol is the network protocol of unix socket.
+	unixProtocol = "unix"
+	apiPath      = "apis/eraser.sh/v1alpha1"
+)
+
+var (
+	// Timeout  of connecting to server (default: 5m).
+	timeout                  = 5 * time.Minute
+	errProtocolNotSupported  = errors.New("protocol not supported")
+	errEndpointDeprecated    = errors.New("endpoint is deprecated, please consider using full url format")
+	errOnlySupportUnixSocket = errors.New("only support unix socket endpoint")
+	log                      = logf.Log.WithName("collector")
+)
+
+type client struct {
+	images  pb.ImageServiceClient
+	runtime pb.RuntimeServiceClient
+}
+
+type Client interface {
+	listImages(context.Context) ([]*pb.Image, error)
+}
+
+func (c *client) listImages(ctx context.Context) (list []*pb.Image, err error) {
```

Comment:
looks like several of these functions are duplicated from eraser, can we refactor to share?